plumbing/object/tree_test.go

```diff
@@ -48,6 +48,22 @@ func (s *TreeSuite) TestType(c *C) {
 	c.Assert(s.Tree.Type(), Equals, plumbing.TreeObject)
 }
 
+func (s *TreeSuite) TestTree(c *C) {
```

Comment:
Could you add a test for Tree(path) when path points to a Blob instead of a Tree?